test/e2e/scheduler_predicates.go

```diff
@@ -0,0 +1,368 @@
+	}
+	return result
+}
+
+var _ = Describe("SchedulerPredicates", func() {
+	var c *client.Client
+	var nodeList *api.NodeList
+	var nodeCount int
+	var totalPodCapacity int64
+	var RCName string
+	var ns string
+	var uuid string
+
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		expectNoError(err)
+		nodeList, err = c.Nodes().List(labels.Everything(), fields.Everything())
+		expectNoError(err)
+		nodeCount = len(nodeList.Items)
+		Expect(nodeCount).NotTo(BeZero())
+
+		err = deleteTestingNS(c)
+		expectNoError(err)
+
+		nsForTesting, err := createTestingNS("sched-pred", c)
+		ns = nsForTesting.Name
+		expectNoError(err)
+		uuid = string(util.NewUUID())
+	})
+
+	AfterEach(func() {
+		rc, err := c.ReplicationControllers(ns).Get(RCName)
+		if err == nil && rc.Spec.Replicas != 0 {
+			By("Cleaning up the replication controller")
+			err := DeleteRC(c, ns, RCName)
+			expectNoError(err)
+		}
+
+		By(fmt.Sprintf("Destroying namespace for this suite %v", ns))
+		if err := c.Namespaces().Delete(ns); err != nil {
+			Failf("Couldn't delete ns %s", err)
+		}
+	})
+
+	// This test verifies that max-pods flag works as advertised. It assumes that cluster add-on pods stay stable
+	// and cannot be run in parallel with any other test that touches Nodes or Pods. It is so because to check
+	// if max-pods is working we need to fully saturate the cluster and keep it in this state for few seconds.
+	It("validates MaxPods limit number of pods that are allowed to run.", func() {
+		totalPodCapacity = 0
+
+		for _, node := range nodeList.Items {
+			podCapacity, found := node.Status.Capacity["pods"]
+			Expect(found).To(Equal(true))
+			totalPodCapacity += podCapacity.Value()
+			Logf("Node: %v", node)
+		}
+
+		pods, err := c.Pods(api.NamespaceAll).List(labels.Everything(), fields.Everything())
+		expectNoError(err)
+		currentlyRunningPods := len(pods.Items)
+		podsNeededForSaturation := int(totalPodCapacity) - currentlyRunningPods
+
+		By(fmt.Sprintf("Starting additional %v Pods to fully saturate the cluster max pods and trying to start another one", podsNeededForSaturation))
+
+		startPods(c, podsNeededForSaturation, ns, "maxp", api.Pod{
+			TypeMeta: api.TypeMeta{
+				Kind: "Pod",
+			},
+			ObjectMeta: api.ObjectMeta{
+				Name:   "",
+				Labels: map[string]string{"name": ""},
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:  "",
+						Image: "gcr.io/google_containers/pause:go",
+					},
+				},
+			},
+		})
+
+		_, err = c.Pods(ns).Create(&api.Pod{
+			TypeMeta: api.TypeMeta{
+				Kind: "Pod",
+			},
+			ObjectMeta: api.ObjectMeta{
+				Name:   "additional-pod",
+				Labels: map[string]string{"name": "additional"},
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:  "additional-pod",
+						Image: "gcr.io/google_containers/pause:go",
+					},
+				},
+			},
+		})
+		expectNoError(err)
+		// Wait a bit to allow scheduler to do its thing
+		time.Sleep(10 * time.Second)
```

Comment:
you can just watch with a timeout instead of time.sleep followed by a list (may not be worth the effort)